### ISSUE TITLE
The process has forked and you cannot use this CoreFoundation functionality safely

### ISSUE THREAD

**Issue by ivan-avalos** [timvisee/prs#1]
I'm using `prs-cli 0.1.2` in macOS Big Sur, and after `prs copy`'ing, the clipboard is cleared and the following message starts getting printed multiple times on the terminal:

```
The process has forked and you cannot use this CoreFoundation functionality safely. You MUST exec().
Break on __THE_PROCESS_HAS_FORKED_AND_YOU_CANNOT_USE_THIS_COREFOUNDATION_FUNCTIONALITY___YOU_MUST_EXEC__() to debug.
```

Let me know if you need more information!

**Comment by timvisee** [comment]
Thanks for the report!

- Does this come up after 20 seconds? 
- Is the clipboard actually cleared (reverted), or does it fail?

This must be caused by the way the delayed clipboard clearing is set up. It might not be supported on macOS. I might have to find a different way to do this then.

**Comment by ivan-avalos** [comment]
This seems to occur after the 20 seconds, indeed, and the clipboard isn't being cleared. Perhaps `password-store` code could give you a hint: https://git.zx2c4.com/password-store/tree/src/password-store.sh

**Comment by ivan-avalos** [comment]
Is there a way to disable clipboard clearing for now? Until this issue if fixed.

**Comment by ivan-avalos** [comment]
Oh, I closed accidentally.

**Comment by timvisee** [comment]
I'm looking into a proper way to achieve this at the moment. The current approach uses forking, as you've noticed, to easily spawn a background task that outlives the main `prs` thread to support clearing after it quites. As you've noticed this method isn't supported on macOS.

> This seems to occur after the 20 seconds, indeed, and the clipboard isn't being cleared. Perhaps password-store code could give you a hint: https://git.zx2c4.com/password-store/tree/src/password-store.sh#n155

Thanks for the hint. Thought I don't think this nicely supports reverting clipboard contents after clearing, it may be a nice temporary fix.

> Is there a way to disable clipboard clearing for now? Until this issue if fixed.

Not at the moment. If you're calling through an alias or function, you may be able to hide the error with:

```bash
prs copy 2>&1 /dev/null
```

Otherwise, would a `--no-clear` flag suffice? That's something I can quickly add.

**Comment by ivan-avalos** [comment]
> Otherwise, would a `--no-clear` flag suffice? That's something I can quickly add.

That would be nice!

**Comment by timvisee** [comment]
Just noticed this can already be achieved through the timeout option instead. A timeout of 0 won't do the clearing.

Simply supply `--timeout 0` or `-t0`.

**Comment by timvisee** [comment]
I experimented with a new clipboard timeout approach. I've not been able to test it on a macOS machine yet, but it does seem to work fine on Linux.

It would be awesome if you could give the latest build including this change for macOS a try:

https://gitlab.com/timvisee/prs/-/jobs/910479654/artifacts/browse

**Comment by timvisee** [comment]
I've just tested these changes on macOS, and it seems to work fine.

I've released these as part of `v0.1.3`. This version should be available once the release pipeline succeeds.

- [Release pipeline](https://gitlab.com/timvisee/prs/-/pipelines/229698194)
- [Downloads](https://gitlab.com/timvisee/prs#install)
- [Changelog](https://gitlab.com/timvisee/prs/-/blob/master/CHANGELOG.md#013-2020-12-14)

Closing this now. Feel free to open this again if you're still experiencing these issues with `v0.1.3`.

Also, since `prs` is in early development, I'd love to get some feedback.
